src/method/method_def.hpp

```diff
@@ -128,15 +137,13 @@ struct method_sig_node
   /* method signature */
   METHOD_SIG *next;
   char *method_name;		/* method name */
+  char *auth_name;              /* auth name (owner or caller) */
   METHOD_TYPE method_type;	/* instance or class method */
   int num_method_args;		/* number of arguments */
   int *method_arg_pos;		/* arg position in list file */
 
-  union
-  {
-    char *class_name;		/* class name for the class method */
-    METHOD_ARG_INFO arg_info;  /* argument info for javasp's server-side calling */
-  };
+  char *class_name;		/* class name for the class method */
+  METHOD_ARG_INFO *arg_info;    /* argument info */
```

Comment:
Is arg_info really necessary to be dynamic? Why did you declare it as a pointer?

Reply:
If the method_def is for C method, arg_info is not used.

Reply:
I understood that **arg_info** would be NULL except for C methods, right?

Reply:
Yes.